File: git_repo/test/js/led2.js
var args = process.argv;
var SYSFS_USR3= "/sys/class/leds/beaglebone:green:usr3"

function write_to_LED(dir, file, value) {
	var fs = require('fs');
	try {
		fs.writeFileSync(dir+file, value);	
	}
	catch (error) {
	console.log("writing to file is failed:" + dir+file);
	}

}


switch (args[2]) {
	case 'on':
		write_to_LED(SYSFS_USR3, "/trigger", "none");
		write_to_LED(SYSFS_USR3, "/brightness", 1);
		console.log("LED is turned ON");
		break;
	case 'off':
		write_to_LED(SYSFS_USR3, "/brightness", 0);
		console.log("LED is turned OFF");
		break;
	case 'flash':
		write_to_LED(SYSFS_USR3, "/trigger", "timer");
		write_to_LED(SYSFS_USR3, "/delay_on", 50);
		write_to_LED(SYSFS_USR3, "/delay_off", 50);
		console.log("LED is FLASHing");
		break;
	default:
		console.log("Usage is:");
		console.log("node led2 on|off|flash|help");
}

File: git_repo/test/c/led2.c
#include <stdio.h>
#define SYSFS_USR3 "/sys/class/leds/beaglebone:green:usr3"

void write_to_LED(char filename[], char value[]);


int main(int agrc, char* argv[])
{
	FILE *fp1, *fp2, *fp3, *fp4;
	if(strcmp(argv[1], "on") == 0){
		write_to_LED("/trigger", "none");
		write_to_LED("/brightness", "1");
	}
	else if(strcmp(argv[1], "off") == 0)
		write_to_LED("/brightness", "0");
	else if(strcmp(argv[1], "flash") == 0){
		write_to_LED("/trigger", "timer");
		write_to_LED("/delay_on", "50");
		write_to_LED("/delay_off", "50");
	}
	else if(strcmp(argv[1], NULL) == 0){
		printf("Usage is \n");
		printf("./led2 on|off|flash\n");
		return 0;
	}
}

void write_to_LED(char file[], char value[]){
	FILE* fp;
	char pathname[100];
	sprintf(pathname, SYSFS_USR3 "%s", file);
	fp = fopen(pathname, "w+");
	fprintf(fp, "%s", value);
	fclose(fp);
}
